### DISCUSSION TITLE
Improve functions debbugging

### DISCUSSION THREAD

**Discussion by Olyno** [supabase/supabase#8244] [Feature Requests]
Hi 👋🏻 
A lot of people have some difficulties to debug functions.

From **Lior539** on Discord:

> I find it incredibly difficult to debug any issues with Postgres function. The Postgres logs on Supabase show me what the error was, but there's no help beyond. This makes it a guessing game when it comes to fixing your postgres code. Ultimately I ended wasting a ton of time debugging issues with postgres functions and my project is still very simple. I shudder to think what would happen as it grows more complex

I think it can be interesting to improve that part.

**Comment by NanoCode012** [comment]
I think this is a great feature. I hope this feature could also be extended towards local development.

In my case, there are no logs to see on the development Studio, so I had to manually `/bin/bash` into the container, find the logs location, AND `grep` the correct piece of log.. (realtime generated lots of logs..)

**Comment by anngbaum** [comment]
We've leveraged RPC functions pretty heavily, and also have had a lot of difficulty debugging. This became meaningfully more complex when we layered in RLS and some wonkiness in the client auth provider (discussed in other tasks).

The biggest logging challenge in my experience was connecting the dots between `edge_logs`, which contained response data, and the `postgres_logs`, which had the information about the headers and authentication status. I discussed the problem in more detail [here](https://discord.com/channels/839993398554656828/989219603399974923/993955089800773824) – a team member suggested that we should be able to use cf_ray to connect between tables, but I have not actually been able to do that successfully.

Either a clearer template to join between `edge_logs` and `postgres_logs`, or better yet combining response data with header authentication data would be a huge help.

**Reply by anngbaum** [reply]
@steve-chavez curious if the work [here](https://github.com/PostgREST/postgrest/issues/2506#issuecomment-1485089967) would address the comment above – it's possible that I'm misunderstanding the `server-trace-header`, but if it does create a link between `edge_logs` and `request_logs` that'd be a huge help from my perspective for debugging user-generated reports!

**Reply by steve-chavez** [reply]
@anngbaum Yes, I believe `server-trace-header` can help with that.

PostgREST v11(which contains the feature) is still not deployed though, but should be soon. You can subscribe [here](https://github.com/supabase/postgres/pull/628) for knowing when it's ready.
